File: pyscraper/webscraper/apiscraper/redditscrape.py
from apiscraper.apiscraper import ApiScraper
from praw import Reddit
from os import environ as env
from dotenv import load_dotenv

class RedditApi(ApiScraper):

    def __init__(self, subreddit):
        load_dotenv()
        self.subreddit = subreddit
        self.reddit = Reddit(
            client_id=env['REDDIT_CLIENT_ID'],
            client_secret=env['REDDIT_SECRET'],
            password=env['REDDIT_PASSWORD'],
            user_agent='Mozilla/5.0 (Windows NT 6.1; WOW64; rv:12.0) Gecko/20100101 Firefox/12.0',
            username=env['REDDIT_USERNAME']
        )

    def get(self):
        posts = self.reddit.subreddit(self.subreddit).hot(limit=10)
        return posts

    def collect(self, data):
        for post in data:
            yield {
                'title':  post.title,
                'url': self.reddit.config.reddit_url + post.permalink
            }

File: Deprecated/Strategies/RedditScrape.js
import Snoowrap from 'snoowrap';
import error from '../utils/error.js';
import BaseError from '../utils/baseError.js';
import codes from '../utils/error.js'
import dotenv from 'dotenv'
dotenv.config();


export class RedditScrape {

    setParams(paramList) {
        return {subreddit : paramList[1]}
    }

    async execute (params) {

        var request = null;

        try{
            request = new Snoowrap({
               userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36',
               clientId: process.env.REDDIT_CLIENT_ID,
               clientSecret: process.env.REDDIT_SECRET,
               username: process.env.REDDIT_USERNAME,
               password: process.env.REDDIT_PASSWORD
           })
       }catch(err) {
          Logger.error(err)
           throw new BaseError("Reddit Initialization Error", codes.SERVER_ERROR, err.message)
       }

       let {subreddit} = params

        let itemArr = []

       try{

            const posts = await request.getSubreddit(subreddit).getNew()

            itemArr = posts.map(post => ({
                title: encodeURIComponent(post.title),
                url: encodeURIComponent(`https://www.reddit.com${post.permalink}`)
            }))

            return itemArr

       }catch(err) {
            Logger.error(err)
            throw new BaseError(err.name, codes.SERVER_ERROR, err.message)
       }finally {
            Logger.info(`Scraper Completed at ${new Date()} for subreddit: ${subreddit}`)
       }

    }
}